sdks/typescript/src/instrumentors/anthropic-otel-instrumentation.ts

```diff
@@ -0,0 +1,1099 @@
+      shimmer.unwrap(AnthropicClass.prototype.messages, 'create');
+    }
+
+    return moduleExports;
+  }
+
+  private _wrapMessagesCreate(prototype: any): void {
+    const instrumentation = this;
+
+    // Check if prototype chain has messages.create
+    if (!prototype.messages) {
+      // Try to use Proxy if lazy getter pattern is detected
+      if ((this._config as AnthropicInstrumentationConfig).fallbackToProxy) {
+        logger.info('[AnthropicInstrumentation] messages not found, applying Proxy fallback');
+        this._applyProxyToMessagesGetter(prototype);
+        return;
+      }
+      logger.error('[AnthropicInstrumentation] messages not found on prototype');
+      return;
+    }
+
+    if (!prototype.messages.create) {
+      logger.error('[AnthropicInstrumentation] messages.create not found');
+      return;
+    }
+
+    if (isWrapped(prototype.messages.create)) {
+      logger.debug('[AnthropicInstrumentation] messages.create already wrapped');
+      return;
+    }
+
+    shimmer.wrap(prototype.messages, 'create', (original: Function) => {
+      return async function (this: any, ...args: any[]) {
+        // Skip if shutting down
+        if (isSDKShuttingDown()) {
+          return original.apply(this, args);
+        }
+
+        const [params] = args;
+        const tracer = instrumentation.tracer;
+        const config = instrumentation._config as AnthropicInstrumentationConfig;
+
+        // Get the active context to preserve parent-child relationships
+        const activeContext = context.active();
+        const parentSpan = trace.getSpan(activeContext);
+
+        if (parentSpan) {
+          logger.debug('[AnthropicInstrumentation] Found parent span:', {
+            parentSpanId: parentSpan.spanContext().spanId,
+            parentTraceId: parentSpan.spanContext().traceId,
+            parentSpanName: (parentSpan as any).name || 'unknown',
+          });
+        } else {
+          logger.debug('[AnthropicInstrumentation] No parent span found in context');
+        }
+
+        logger.info('[AnthropicInstrumentation] messages.create called, creating span');
+
+        // Start span using the OTel way with active context
+        const span = tracer.startSpan(
+          `anthropic.messages ${params?.model || 'unknown'}`,
+          {
+            kind: SpanKind.CLIENT,
+            attributes: {
+              [SEMATTRS_GEN_AI_SYSTEM]: 'anthropic',
+              [SEMATTRS_GEN_AI_REQUEST_MODEL]: params?.model,
+              [SEMATTRS_GEN_AI_REQUEST_TEMPERATURE]: params?.temperature,
+              [SEMATTRS_GEN_AI_REQUEST_MAX_TOKENS]: params?.max_tokens,
+              [SEMATTRS_GEN_AI_REQUEST_TOP_P]: params?.top_p,
+              'gen_ai.request.top_k': params?.top_k,
+              [SEMATTRS_GEN_AI_OPERATION_NAME]: 'chat',
+              'lilypad.type': 'llm',
+              'server.address': 'api.anthropic.com',
+            },
+          },
+          activeContext,
```

Comment:
Looks like this span creation logic is duplicated 3 times in this file, can we refactor to DRY?